test/jdk/jdk/crac/TestInetAddress.java

```diff
@@ -0,0 +1,166 @@
+        try {
+            Future<?> completed = startTestProcess();
+            checkpointTestProcess();
+            completed.get(5, TimeUnit.SECONDS);
+            startRestoredProcess();
+        } finally {
+            ensureContainerDead();
+            DockerTestUtils.removeDockerImage(imageName);
+        }
+    }
+
+    private static Future<?> startTestProcess() throws Exception {
+        ensureContainerDead();
+
+        List<String> cmd = new ArrayList<>();
+        cmd.add(Container.ENGINE_COMMAND);
+        cmd.addAll(Arrays.asList("run", "--rm", "--add-host", TEST_HOSTNAME + ":192.168.12.34"));
+        cmd.addAll(Arrays.asList("--volume", Utils.TEST_CLASSES + ":/test-classes/"));
+        cmd.addAll(Arrays.asList("--volume", "cr:/cr"));
+        cmd.addAll(Arrays.asList("--volume", CRAC_CRIU_PATH + ":/criu"));
+        cmd.addAll(Arrays.asList("--env", "CRAC_CRIU_PATH=/criu"));
+        cmd.addAll(Arrays.asList("--entrypoint", "bash"));
+        // checkpoint-restore does not work without this: TODO fine-grained --cap-add
+        cmd.add("--privileged");
+        cmd.addAll(Arrays.asList("--name", CONTAINER_NAME));
+        cmd.add(imageName);
+        // Checkpointing does not work for PID 1, therefore we add an intermediary bash process
+        List<String> javaCmd = new ArrayList<>();
+        javaCmd.addAll(Arrays.asList("/jdk/bin/java", "-cp /test-classes/", "-XX:CRaCCheckpointTo=/cr"));
+        javaCmd.addAll(Arrays.asList(Utils.getTestJavaOpts()));
+        javaCmd.addAll(Arrays.asList("ResolveInetAddress", TEST_HOSTNAME, "/second-run"));
+        cmd.addAll(Arrays.asList("-c", String.join(" ", javaCmd) + "; echo i-am-here-to-force-child-process"));
+
+        System.err.println("Running: " + String.join(" ", cmd));
+
+        CompletableFuture<?> firstOutputFuture = new CompletableFuture<Void>();
+        Future<?> completed = executeWatching(cmd, line -> {
+            System.out.println("OUTPUT: " + line);
+            if (line.equals("192.168.12.34")) {
+                firstOutputFuture.complete(null);
+            }
+        }, error -> {
+            System.err.println("ERROR: " + error);
+            firstOutputFuture.cancel(false);
+        });
+        firstOutputFuture.get(10, TimeUnit.SECONDS);
+        return completed;
+    }
+
+    private static void ensureContainerDead() throws Exception {
+        // ensure the container is not running, ignore if not present
+        DockerTestUtils.execute("docker", "kill", CONTAINER_NAME).getExitValue();
+    }
+
+    private static void checkpointTestProcess() throws Exception {
+        int javaPid = DockerTestUtils.execute("docker", "exec", CONTAINER_NAME, "/jdk/bin/jps").asLines().stream()
+                .filter(line -> !line.isBlank() && !line.contains("Jps"))
+                .mapToInt(line -> Integer.parseInt(line.split("\\s")[0]))
+                .min().orElseThrow();
```

Comment:
Not something really need to fix, but it's possible to specify just e.g. class-name and not parse jps output: https://github.com/openjdk/crac/blob/master/src/jdk.jcmd/share/man/jcmd.1#L58